src/mlpack/core/optimizers/parallel_sgd/sparse_svm_function.hpp

```diff
@@ -0,0 +1,55 @@
+/**
+ * @file sparse_svm_function.hpp
+ * @author Shikhar Bhardwaj
+ *
+ * Implementation of the hinge loss function for training a sparse SVM with the
+ * parallel SGD algorithm
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_PARALLEL_SGD_SPARSE_SVM_HPP
+#define MLPACK_CORE_OPTIMIZERS_PARALLEL_SGD_SPARSE_SVM_HPP
+#include <mlpack/prereqs.hpp>
+
+class SparseSVMLossFunction{
+ public:
+  //! Nothing to do for the default constructor.
+  SparseSVMLossFunction() = default;
+
+  //! Member initialization constructor.
+  SparseSVMLossFunction(arma::sp_mat& dataset, arma::vec& labels);
+
+  //! Evaluate a function.
+  double Evaluate(const arma::mat& weights, size_t id);
+
+  //! Evaluate the gradient of a function.
+  void Gradient(const arma::mat& weights, size_t id, arma::sp_mat& gradient);
```

Comment:
If this is all that is necessary for the function that interfaces with Hogwild, do you think we could adapt our existing `LogisticRegression` and `SoftmaxRegression` implementations to handle this?  I think all that would need to change would be that the type of the gradient should be templatized.